src/static_array.cr

```diff
@@ -67,31 +111,74 @@ struct StaticArray(T, N)
     indexes.map { |index| self[index] }
   end
 
+  # Calls the given block at the given index.
+  # Raises `IndexError` if trying to set an element outside the array's range.
+  #
+  # ```
+  # array = StaticArray(Int32, 3).new { |i| i+1 } # => [1, 2, 3]
+  # array.update(1) { |x| x * 2 } # => [1, 4, 3]
+  # array.update(5) { |x| x * 2 } # => IndexError
+  # ```
   def update(index : Int)
     index = check_index_out_of_bounds index
     to_unsafe[index] = yield to_unsafe[index]
   end
 
+  # Returns the size of `self`
+  #
+  # ```
+  # array = StaticArray(Int32, 3).new { |i| i+1 }
+  # array.size # => 3
+  # ```
   def size
     N
   end
 
+  # Returns all the elements of the array by the given value
+  # 
+  # ```
+  # array = StaticArray(Int32, 3).new { |i| i+1 }
+  # array.[]= 2 # => [2, 2, 2]
+  #
+  # ```
   def []=(value : T)
     size.times do |i|
       to_unsafe[i] = value
     end
   end
 
+  # Returns an array with all the elements in the collection randomized
+  # Mutates the array
+  #
+  # ```
+  # array = StaticArray(Int32, 3).new { |i| i+1 } # => [1, 2, 3]
+  # array.shuffle! # => [1, 3, 2]
+  # array.shuffle! # => [3, 2, 1]
+  # array # => [3, 2, 1]
+  # ```
   def shuffle!
     to_unsafe.shuffle!(size)
     self
   end
-
+  
+  # Invokes the given block for each element of `self`, replacing the element
+  # with the value returned by the block. Returns `self`.
+  #
+  # ```
+  # array = StaticArray(Int32, 3).new { |i| i+1 }
+  # array.map! { |x| x*x } # => [1, 4, 9]
+  # ```
   def map!
     to_unsafe.map!(size) { |e| yield e }
     self
   end
 
+  # Returns an array with all the elements in the collection reversed.
```

Comment:
Should be something like "Reverses the elements of this array in-place, then returns `self`"